airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/source.py

```diff
@@ -140,6 +144,16 @@ def _make_authenticator(config: Mapping[str, Any]):
 
     @staticmethod
     def _choose_profiles(config: Mapping[str, Any], profiles: List[Profile]):
-        if not config.get("profiles"):
+        filtered_profiles = []
+        if config.get("profiles"):
+            for profile in list(filter(lambda profile: profile.profileId in config["profiles"], profiles)):
+                filtered_profiles.append(profile)
+        if config.get("marketplace_ids"):
+            for profile in list(filter(lambda profile: profile.accountInfo.marketplaceStringId in config["marketplace_ids"], profiles)):
+                if profile not in filtered_profiles:
+                    filtered_profiles.append(profile)
+
+        if config.get("profiles") or config.get("marketplace_ids"):
+            return filtered_profiles
+        else:
             return profiles
```

Comment:
```suggestion
        requested_profiles = config.get("profiles", [])
        requested_marketplace_ids = config.get("marketplace_ids", [])
        return [profile for profile in profiles if profile in requested_profiles or profile.accountInfo.marketplaceStringId in requested_marketplace_ids]
```

Reply:
Also in `_choose_profiles` method we can re-name `profiles` argument to `available_profiles`. This makes the code more readable.

Reply:
I modified this to return `available_profiles` if there is no filter in place.